Buildings/Experimental/DHC/EnergyTransferStations/Cooling/DirectControlled.mo

```diff
@@ -0,0 +1,293 @@
+  parameter Modelica.Fluid.Types.Dynamics energyDynamics=Modelica.Fluid.Types.Dynamics.FixedInitial
+    "Type of energy balance: dynamic (3 initialization options) or steady state";
+  parameter Modelica.Units.SI.PressureDifference[3] dp_nominal=500*{1,-1,1}
+    "Nominal pressure drop in pipe junctions";
+  Buildings.Fluid.Sensors.TemperatureTwoPort senTDisSup(
+    redeclare final package Medium=MediumSer,
+    final m_flow_nominal=mDis_flow_nominal)
+    "District supply temperature sensor"
+    annotation (Placement(transformation(extent={{-180,-290},{-160,-270}})));
+  Buildings.Fluid.Sensors.MassFlowRate senMasFlo(
+    redeclare final package Medium=MediumSer)
+    "District supply mass flow rate sensor"
+    annotation (Placement(transformation(extent={{-120,-290},{-100,-270}})));
+  Buildings.Fluid.FixedResistances.Junction jun(
+    redeclare final package Medium=MediumSer,
+    final energyDynamics=energyDynamics,
+    final m_flow_nominal={mDis_flow_nominal,-mBui_flow_nominal,mByp_flow_nominal},
+    final dp_nominal=dp_nominal)
+    "Bypass junction"
+    annotation (Placement(transformation(extent={{-60,-270},{-40,-290}})));
+  Buildings.Fluid.Sensors.TemperatureTwoPort senTDisRet(
+    redeclare final package Medium=MediumSer,
+    final m_flow_nominal=mDis_flow_nominal)
+    "District return temperature sensor"
+    annotation (Placement(transformation(extent={{30,210},{50,190}})));
+  Buildings.Fluid.FixedResistances.Junction spl(
+    redeclare final package Medium=MediumSer,
+    final energyDynamics=energyDynamics,
+    final m_flow_nominal={mBui_flow_nominal,-mDis_flow_nominal,-mByp_flow_nominal},
+    final dp_nominal=dp_nominal)
+    "Bypass junction, splitter"
+    annotation (Placement(transformation(extent={{-60,190},{-40,210}})));
+  Buildings.Fluid.Actuators.Valves.TwoWayEqualPercentage conVal(
+    redeclare final package Medium=MediumSer,
+    final m_flow_nominal=mDis_flow_nominal,
+    final dpValve_nominal=dpConVal_nominal,
+    use_inputFilter=true,
+    riseTime(displayUnit="s") = 60)
+    "Control valve"
+    annotation (Placement(transformation(extent={{-10,210},{10,190}})));
+  Buildings.Fluid.Sensors.TemperatureTwoPort senTBuiRet(redeclare final package
+      Medium = MediumSer, final m_flow_nominal=mBui_flow_nominal)
+    "Building return temperature sensor"
+    annotation (Placement(transformation(extent={{-220,210},{-200,190}})));
+  Modelica.Fluid.Valves.ValveIncompressible cheVal(
+    redeclare final package Medium=MediumSer,
+    final allowFlowReversal=false,
+    final dp_nominal=dpCheVal_nominal,
+    final m_flow_nominal=mByp_flow_nominal,
+    final filteredOpening=true,
+    riseTime(
+      displayUnit="s")=60,
+    final checkValve=true)
+    "Check valve (backflow preventer)"
+    annotation (Placement(transformation(extent={{10,-10},{-10,10}},
+        rotation=90,
+        origin={-50,-10})));
+  Modelica.Blocks.Math.Add dTdis(
+    final k1=-1,
+    final k2=+1)
```

Comment:
may not need to redefine k2=+1, as in default it is 1.

Reply:
By putting a "final" prefix for k2=+1, my intention is to keep k2 unchanged. Does that make sense?

Reply:
I think there is no need to have the `final k2=+1`.